swiftspec/core.py

```diff
@@ -281,6 +281,13 @@ async def _rm(self, path, recursive=False, batch_size=None, **kwargs):
             nofiles=True,
         )
 
+    async def _get_file(self, rpath, lpath, **kwargs):
+        if self.isdir(rpath):
+            return os.makedirs(lpath, exist_ok=True)
```

Comment:
I'm not sure if we need this (but probably we do...). If we do, this should be async, otherwise this call will block other calls, which might also be the reason for a deadlock.

```suggestion
        if await self._isdir(rpath):
            return os.makedirs(lpath, exist_ok=True)
```